### ISSUE TITLE
Single access point for experiments

### ISSUE THREAD

**Issue by ADGEfficiency** [ADGEfficiency/energy_py#34]
Currently provide and document two ways to run experiments - one using a dictionary and second using config files.  Will remove the dictionary experiment and offer only config files.

**Comment by honglinwen** [comment]
HI, I am excited to find your repo. I followed your starting step, using `python experiment.py example dqn`, but it arose a KeyError: 'example'. Could you please help me to fix the error? thanks.

**Comment by ADGEfficiency** [comment]
Hi - thanks for the interest.

What do you have in the folder `energy_py/energy_py/experiments/datasets/example` ?

**Comment by honglinwen** [comment]
I cloned this repo to my computer. That folder contains two csv files, namely `observation.csv` and `state.csv`. I guessed that is a demo and ran using that command.

**Comment by ADGEfficiency** [comment]
You are right that this is a demo.  I can only reproduce your error if I remove/rename the `example` folder from `energy_py/experiments/datasets`

Try

```
$ cd energy_py/experiments/datasets

$ python -i __init__.py

>>> registry.keys()

```

After doing this I get `dict_keys(['__pycache__', 'example', 'tempus'])` because I have the example dataset plus one for Tempus work (the pycache shouldn't be in there really :))

**Comment by honglinwen** [comment]
It's so strange. I did this and got `dict_keys(['example'])`. But it still arose that KeyError when I ran the demo. :(

**Comment by ADGEfficiency** [comment]
@honglinwen - I just pushed up a change 

```
def get_dataset_path(dataset_name):
    return registry[str(dataset_name)]
```

I also added some protection into the `BaseEnv` class init

Theory being perhaps it's the encoding of the string?

**Comment by samarthiith** [comment]
I tried running it on windows and mac, both showing the same error.

**Comment by tomharvey** [comment]
When the package is installed, the current directory of the datasets folder is the installed version.

When using develop, the current directory of the datasets folder is the repo version.

The installed version does not include the `example` directory:
```
ls  ~/.pyenv/versions/energy_py/lib/python3.6/site-packages/energy_py-0.2-py3.6.egg/energy_py/experiments/datasets

__init__.py	__pycache__
```

In `setup.py` the packages are being defined as `packages=find_packages(exclude=['tests', 'tests.*']),` . The `find_packages` will only include directories which have `__init__.py` files in them - which `experiments/datasets/example` does not have.

You're using `package_data = {'':['*.csv']}` to add CSV files but I don't think it's doing what you want.

**Comment by tomharvey** [comment]
I've added a PR which should solve it  by installing the files as package_data - but check https://docs.python.org/2/distutils/setupscript.html#installing-package-data for the difference between `package_data` and `data_files`to determine if this is the correct fix.

**Comment by ADGEfficiency** [comment]
Thanks @tomharvey - the whole reason for doing this complex dataset finding stuff was to avoid this issue (of where the package is installed versus where it was cloned too) - seems it didn't work!  

I'll have a look at this over the next day or so and try to make some changes.   

In the meantime - @honglinwen and @samarthiith - can you run `python setup.py develop` to install the package and see if that fixes your issue?  If it fixes it I will change the readme as a bandaid until a proper fix is in place

**Comment by tomharvey** [comment]
If you don't want the CSV files in the package then passing a path to the dataset into the command would be a better option. Something like `python experiment.py dqn --dataset ~/datasets/example`

**Comment by honglinwen** [comment]
thanks @ADGEfficiency @tomharvey  it works :+1:
